### ISSUE TITLE
`AtomicFile::new` with a relative path like `foo` will panic

### ISSUE THREAD

**Issue by nikomatsakis** [untitaker/rust-atomicwrites#19]
If you do this:

```rust
fn main() {
    AtomicFile::new("foo", OverwriteBehavior::AllowOverwrite).write(|_| {
        Ok::<(),()>(())
    }).unwrap();
}
```

then you get an I/O error:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Internal(Os { code: 2, kind: NotFound, message: "No such file or directory" })', libcore/result.rs:945:5
```

I believe the cause is that `new` invokes `parent()` and then supplies (if there is no parent) the value `p`:

https://github.com/untitaker/rust-atomicwrites/blob/3ad97ac63fc6dc92fc0887d66502e99403dda7e4/src/lib.rs#L84-L87

Probably it should supply the current directory? Or perhaps just `None`?

(My workaround was to invoke `canonicalize` on the path.)

**Comment by untitaker** [comment]
This is really bad, thanks for reporting. The intent is to put the temporary file in the same directory as the target file to ensure they are stored on the same physical volume. Otherwise rename will not be atomic

**Comment by untitaker** [comment]
Well, today I learned how `parent()` actually works. Thanks, released 0.1.5.
